runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -116,20 +116,18 @@ public void invoke() {
                 Layout requestLayout = new Layout(runtime.getLayoutView().getLayout());
                 UUID workflowId = sendRequest(requestLayout);
                 ManagementClient orchestrator = getOrchestrator(requestLayout);
-
                 waitForWorkflow(workflowId, orchestrator, timeout, pollPeriod);
+            } catch (NetworkException | TimeoutException e) {
+                log.warn("WorkflowRequest: Error while running {} on attempt {}, cause {}", this, x, e);
+            }
 
-                for (int y = 0; y < runtime.getParameters().getInvalidateRetry(); y++) {
-                    runtime.invalidateLayout();
-                    Layout layoutToVerify = new Layout(runtime.getLayoutView().getLayout());
-                    if (verifyRequest(layoutToVerify)) {
-                        log.info("WorkflowRequest: Successfully completed {}", this);
-                        return;
-                    }
+            for (int y = 0; y < runtime.getParameters().getInvalidateRetry(); y++) {
+                runtime.invalidateLayout();
+                Layout layoutToVerify = new Layout(runtime.getLayoutView().getLayout());
```

Comment:
I know you haven't touched this logic but this defensive copying of layout might be eliminated. It seems that none of the verfiyRequest implementations are mutating the layout (as expected from a verifying method).

Reply:
The layout from line 116 can't be used because it will be the same one before the operation happens, thus missing one try. For example, copy layout1 -> execute operation -> verify layout1. The verification will run on the same layout before the operation, which doesn't make sense.

Reply:
yes, L116 doesn't make sense. I was referring to L126's creation of layoutToVerify. so rather than copying a layout from runtime.getLayoutView().getLayout() and verifying the copy, couldn't we have verifyRequest(runtime.getLayoutView().getLayout())?

Reply:
I see. I think its fine since this code path should execute frequently.